File: pkg/pwrgrep/testdata/fixtures/c/input-reaches-effect.c
#include <stdlib.h>
#include <unistd.h>

int main(int argc, char **argv) {
    char *cmd = argv[1];
    // ruleid: c-input-reaches-effect
    system(cmd);
    // ok: c-input-reaches-effect
    system("ls");
    const char *home = getenv("HOME");
    char *p;
    p = home;
    // ruleid: c-input-reaches-effect
    unlink(p);
    return 0;
}

File: pkg/pwrgrep/testdata/fixtures/java/InputReachesEffect.java
import java.sql.Statement;

public class InputReachesEffect {
    void run() throws Exception {
        String cmd = System.getenv("CMD");
        String full = cmd + " --verbose";
        // ruleid: java-input-reaches-effect
        Runtime.getRuntime().exec(full);
        // ok: java-input-reaches-effect
        Runtime.getRuntime().exec("ls");
    }

    void handle(HttpServletRequest request, Statement stmt) throws Exception {
        String name = request.getParameter("name");
        // ruleid: java-input-reaches-effect
        stmt.executeQuery("SELECT * FROM users WHERE name = '" + name + "'");
        // ok: java-input-reaches-effect
        stmt.executeQuery("SELECT 1");
    }
}
